### PR TITLE
Support MST 1.4 and 2.0 versions

### PR DESCRIPTION
**Pull request by Rabsztok**
Add support for abort function in middleware. For now, both 1.2.1 and >=1.4.0 versions return just `null` as a return value when aborted. Later on, we will add ability to pass custom abort value.

### CODE REVIEW

**Comment by d4rky-pl** [comment]
Please fix the tests first :)

**Comment by d4rky-pl** [comment]
I believe you're not initializing node_modules properly before running the tests, you will probably have to add setup step (for ex. with `beforeAll` and `child_process.spawn`)

**Comment by Rabsztok** [comment]
The problem lies with Travis not installing MST versions from package.json
inside tests directory. I will fix that tomorrow.

czw., 22 mar 2018, 16:10 użytkownik Michał Matyas <notifications@github.com>
napisał:

**Comment by d4rky-pl** [comment]
Yes, but you would have the same problem if you cloned the repository and ran the tests yourself since `node_modules` is gitignored so it's not only CI issue but incorrect tests setup ;)

**Comment by Rabsztok** [comment]
It will fix #18 and #20 issues.

File: .gitignore [hunk 1]
```diff
@@ -2,3 +2,5 @@ node_modules
 .idea
 .DS_Store
 /lib
+__tests__/**/yarn.lock
```
Comment (d4rky-pl) [inline .gitignore]: Since `jest.setup.js` uses npm, this is unnecessary

File: src/middleware.js [hunk 2]
```diff
@@ -3,11 +3,11 @@ const getArborisOpts = call =>
   null
 
 export default function createMiddleware(tracker, logger) {
-  return function(call, next) {
+  return function(call, next, abort) {
     const opts = getArborisOpts(call)
 
     if (opts && opts.type === "async" && tracker.renderLimitReached()) {
-      return false
+      return abort ? abort(null) : null
```
Comment (d4rky-pl) [inline src/middleware.js]: Please change this to `typeof abort === 'function'`

File: src/middleware.js [hunk 3]
```diff
@@ -24,6 +24,7 @@ export default function createMiddleware(tracker, logger) {
             ").\n" +
             "Make sure you're applying Arboris middleware before running any actions."
         )
+        return abort ? abort(null) : null
```
Comment (d4rky-pl) [inline src/middleware.js]: Same as above

File: jest.setup.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+const execSync = require("child_process").execSync
+const mstVersions = ["1.2.1", "1.4.0"]
```
Comment (d4rky-pl) [inline jest.setup.js]: Please add 1.3.0 and 2.0.0

**Review by d4rky-pl — CHANGES_REQUESTED** [review]

**Review by d4rky-pl — APPROVED** [review]

**Pull request merged**